examples/2_advanced/plot_s4_galaxies.py

```diff
@@ -0,0 +1,160 @@
+# with the typical redshift errors of a photometric survey.
+
+# setting up the random number generator:
+rng = np.random.default_rng(seed=42)
+
+# photometric redshift distribution following a Smail distribution
+z = np.linspace(0, 3.0, 1000)
+dndz = glass.observations.smail_nz(z, z_mode=0.9, alpha=2., beta=1.5)
+dndz *= n_arcmin2
+bz = 1.2
+
+# equal density bins:
+nbins = 10
+zedges = glass.observations.equal_dens_zbins(z, dndz, nbins=nbins)
+bin_nz = glass.observations.tomo_nz_gausserr(z, dndz, sigma_z0, zedges)
+
+# %%
+# Plotting the overall redshift distribution and the
+# distribution for each of the equal density tomographic bins
+plt.figure()
+plt.title("Stage IV Space Telescope - Photometric Distribution: equal density bins")
+sum_nz = np.zeros_like(bin_nz[0])
+for nz in bin_nz:
+    plt.fill_between(z, nz, alpha=0.5)
+    sum_nz = sum_nz + nz
+plt.fill_between(z, dndz, alpha=0.2, label='dn/dz')
+plt.plot(z, sum_nz, ls='--', label="Sum of the bins")
+plt.ylabel("dN/dz - gal/arcmin2")
+plt.xlabel("z")
+plt.legend()
+plt.tight_layout()
+plt.show()
+
+# %%
+# Make a visibility map typical of a space telescope survey, seeing both
+# hemispheres, and low visibility in the galactic and ecliptic bands.
+vis = glass.observations.vmap_galactic_ecliptic(nside)
+
+# checking the mask:
+hp.mollview(vis, title='Stage IV Space Survey-like Mask', unit='Visibility')
+plt.show()
+
+# %%
+# generators for the clustering and lensing
+generators = [
+    glass.sim.zspace(0., 3.0001, dz=dz),
+    glass.camb.camb_matter_cl(pars, lmax),
+    glass.matter.lognormal_matter(nside, rng=rng),
+    glass.lensing.convergence(cosmo),
+    glass.lensing.shear(lmax),
+    glass.observations.vis_constant(vis, nside=nside),
+    glass.galaxies.gal_dist_fullsky(z, bin_nz, bz=bz, rng=rng),
+    glass.galaxies.gal_ellip_gaussian(sigma_e, rng=rng),
+    glass.galaxies.gal_shear_interp(cosmo),
+]
+
+# %%
+# Simulation
+# ----------
+# Simulate the galaxies with shears.  In each iteration, get the quantities of interest
+# to build our mock catalogue.
+
+# we will store the catalogue as a dictionary:
+catalogue = {'RA': np.array([]), 'DEC': np.array([]), 'TRUE_Z': np.array([]),
+             'E1': np.array([]), 'E2': np.array([]), 'TOMO_ID': np.array([])}
+
+# iterate and store the quantities of interest for our mock catalogue:
+for shell in glass.sim.generate(generators):
+    # let's assume here that lon lat here are RA and DEC:
+    catalogue['RA'] = np.append(catalogue['RA'], shell['gal_lon'])
+    catalogue['DEC'] = np.append(catalogue['DEC'], shell['gal_lat'])
+    catalogue['TRUE_Z'] = np.append(catalogue['TRUE_Z'], shell['gal_z'])
+    catalogue['E1'] = np.append(catalogue['E1'], shell['gal_ell'].real)
+    catalogue['E2'] = np.append(catalogue['E2'], shell['gal_ell'].imag)
+    catalogue['TOMO_ID'] = np.append(catalogue['TOMO_ID'], shell['gal_pop'])
+
+print(f"Total Number of galaxies sampled: {len(catalogue['TRUE_Z'])}")
```

Comment:
Looks more impressive with the separators!

```suggestion
print(f'Total Number of galaxies sampled: {len(catalogue['TRUE_Z']):,}')
```